synapse/handlers/message.py

```diff
@@ -320,6 +321,25 @@ def create_event(self, requester, event_dict, token_id=None, txn_id=None,
                         target, e
                     )
 
+        if builder.type == EventTypes.Aliases:
+            if "aliases" in builder.content:
+                for alias in builder.content["aliases"]:
+                    if len(alias) > MAX_ALIAS_LENGTH:
+                        raise SynapseError(
+                            400,
+                            ("Can't create aliases longer than"
+                             " %s characters" % MAX_ALIAS_LENGTH),
+                            Codes.INVALID_PARAM,
+                        )
+
+            if self.require_membership_for_aliases:
```

Comment:
I think we should remove this condition, and always enforce it

Reply:
Added comment in e910c29 to describe why we need it.